demos/storybook/stories/app-bar/with-full-config.tsx

```diff
@@ -0,0 +1,120 @@
+        marginLeft: 32,
+        position: 'relative',
+        flexGrow: 1,
+    },
+    expanded: {
+        '& $liner': {
+            top: 64,
+        },
+    },
+    collapsed: {
+        '& $title': {
+            fontSize: '1.25rem',
+            fontWeight: 600,
+        },
+        '& $subtitle': {
+            fontSize: 0,
+        },
+        '& $info': {
+            fontSize: '1rem',
+            fontWeight: 400,
+            marginTop: '-0.25rem',
+        },
+    },
+    marginLeft12: {
+        marginLeft: 12,
+    },
+    negativeMarginLeft12: {
+        marginLeft: -12,
+    },
+    negativeMarginRight12: {
+        marginRight: -12,
+    },
+    negativeMarginRight20: {
+        marginRight: -20,
+    },
+}));
+
+export const withFullConfig = (): StoryFnReactReturnType => {
+    const classes = useStyles();
+    const direction = getDirection();
+    // AppBar props
+    const animationDuration = number('animationDuration', 300);
+    const showBackgroundImage = boolean('show backgroundImage', true);
+    const collapsedHeight = number('collapsedHeight', 64);
+    const expandedHeight = number('expandedHeight', 200);
+    const scrollThreshold = number('scrollThreshold', 136);
+    const variant = select('variant', ['snap', 'collapsed', 'expanded'], 'snap');
+    // ThreeLiner props
+    const title = text('title', 'title');
+    const subtitle = text('subtitle', 'subtitle');
+    const info = text('info', 'info');
+
+    return (
+        <div style={{ display: 'block' }}>
+            <AppBar
+                expandedHeight={expandedHeight}
+                collapsedHeight={collapsedHeight}
+                scrollThreshold={scrollThreshold}
+                animationDuration={animationDuration}
+                backgroundImage={showBackgroundImage ? bgImage : undefined}
+                variant={variant}
+                classes={{ collapsed: classes.collapsed, expanded: classes.expanded }}
+            >
+                <Toolbar>
+                    <IconButton
+                        onClick={action('home icon clicked...')}
+                        color={'inherit'}
+                        edge={'start'}
+                        className={direction === 'rtl' ? classes.marginLeft12 : classes.negativeMarginRight20}
+                    >
+                        <Menu />
+                    </IconButton>
+                    <ThreeLiner
+                        classes={{ title: classes.title, subtitle: classes.subtitle, info: classes.info }}
+                        className={classes.liner}
+                        title={title}
+                        subtitle={subtitle}
+                        info={info}
+                        animationDuration={animationDuration}
+                    />
+                    <div
+                        style={{ display: 'flex', flexDirection: 'row' }}
+                        className={direction === 'rtl' ? classes.negativeMarginLeft12 : classes.negativeMarginRight12}
```

Comment:
JSS is smart enough to flip directional css properties for you automatically (it should work without having to do this).